apps/microsoft-teams/frontend/src/utils/msGraphApi.ts

```diff
@@ -0,0 +1,164 @@
+    super(res.message ?? '');
+    this.status = res.status;
+  }
+}
+
+interface OrganizationalBrandingSuccessResponse {
+  '@odata.context': string;
+  cdnList: string[];
+  faviconRelativeUrl: string;
+  squareLogoRelativeUrl: string;
+}
+
+/**
+ * This class is used to interact with the Microsoft Graph API.
+ * Specifically we use this API to get the organization name and logo to display after a user authenticates
+ * @param instance IPublicClientApplication, comes from useMsal hook
+ * @param account AccountInfo, comes from useMsal hook
+ */
+class MsGraph {
+  readonly baseUrl: string;
+  readonly instance: IPublicClientApplication;
+  readonly account: AccountInfo;
+  readonly tenantId: string;
+
+  constructor(instance: IPublicClientApplication, account: AccountInfo) {
+    this.baseUrl = graphConfig.graphOrgEndpoint;
+    this.instance = instance;
+    this.account = account;
+    this.tenantId = account.tenantId;
+  }
+
+  /**
+   * Look for a valid token in the cache, and if it is close to expiring or does not exist, will automatically try to refresh it for you using the cached refresh token
+   * The silent token requests to Microsoft Entra ID might fail for reasons like a password change or updated Conditional Access policies, then call an interactive method
+   * See: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/acquire-token.md#acquiring-an-access-token
+   * @returns string
+   */
+  private getAccessToken = async (): Promise<string> => {
+    const request = {
+      scopes: ['User.Read'],
+      account: this.account,
+    };
+
+    try {
+      const authSilentResult = await this.instance.acquireTokenSilent(request);
+      return authSilentResult.accessToken;
+    } catch (error) {
+      // if acquiring the token silently fails, then present the user with the popup to get the token
+      if (error instanceof InteractionRequiredAuthError) {
+        const authInteractiveResult = await this.instance.acquireTokenPopup(request);
+        return authInteractiveResult.accessToken;
+      } else {
+        throw error;
+      }
+    }
+  };
+
+  /**
+   * Constructs options object for fetch request
+   * @param headers Headers
+   * @param accessToken string
+   * @returns options: {
+      method: string;
+      headers: Headers;
+    }
+   */
+  private constructOptions = (headers: Headers, accessToken: string) => {
+    const bearer = `Bearer ${accessToken}`;
+    headers.append('Authorization', bearer);
+
+    const options = {
+      method: 'GET',
+      headers,
+    };
+
+    return options;
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private validateResponseStatus = (response: Response, responseJson: any) => {
+    if (response.status >= 400) {
+      const apiErrorResponse = {
+        status: response.status,
+        message: responseJson?.error?.message,
+      };
+
+      throw new MsGraphApiError(apiErrorResponse);
```

Comment:
Does this get caught anywhere?  If not, is there a risk of causing a runtime error that might make MS teams app unusable?